doc/dsl/typebuilding.md

```diff
@@ -16,9 +16,13 @@ game_object my_object {
 }
 ```
 
-Diese Form der Entitätsdefinition erfordert eine Repräsentation der Komponententypen (welche als Java-Klassen / Java-Records im ECS definiert sind) im DSL Typsystem und eine Verbindung des DSL Typen mit der Java-Klasse, dessen DSL-Equivalent er ist.
+Diese Form der Entitätsdefinition erfordert eine Repräsentation der Komponententypen (welche als
+Java-Klassen / Java-Records im ECS definiert sind) im DSL Typsystem und eine Verbindung des
+DSL Typen mit der Java-Klasse, dessen DSL-Equivalent er ist.
+Das folgende Diagramm stellt dar, wie eine Java-Klasse auf der DSL-Seite als `AggregateType`
```

Comment:
Wieso ist alles in "fett" gedruckt? Oder sieht das nur auf dem Bildschirm so aus?

Auf der DSL-Seite verstehe ich die Kanten nicht: Soll das eine Vererbungsbeziehung sein (dann fehlen Pfeile)? Soll das eine Aggregation/Komposition sein (dann fehlen Rauten)? Oder soll das wirklich nur eine "wir kennen uns gegenseitig"-Beziehung sein (glaube ich eher nicht?).

Links in der Java-Seite ist es eine Klassendiagramm, auf der rechten (DSL-) Seite scheinen es bereits konkrete Objekte zu sein? (Werden die dann nicht nochmal extra unterstrichen, um das von den Klassen abzugrenzen?)

Reply:
> Wieso ist alles in "fett" gedruckt? Oder sieht das nur auf dem Bildschirm so aus?

Das sieht wahrscheinlich nur so aus. In umlet hab ich auf jeden Fall nichts "fett" markiert...

 

> Auf der DSL-Seite verstehe ich die Kanten nicht: Soll das eine Vererbungsbeziehung sein (dann fehlen Pfeile)? Soll das eine Aggregation/Komposition sein (dann fehlen Rauten)? Oder soll das wirklich nur eine "wir kennen uns gegenseitig"-Beziehung sein (glaube ich eher nicht?).

Das soll ein Objekt-Diagramm sein, um die Beziehung zwischen den konkreten Objekten darstellen, die der `TypeBuilder` zur Repräsentation der Klasse auf der Java Seite macht. Aber du hast Recht, eigentlich werden die Namen dann unterstrichen.

Reply:
Ah.

Reply:
> > Wieso ist alles in "fett" gedruckt? Oder sieht das nur auf dem Bildschirm so aus?
> 
> Das sieht wahrscheinlich nur so aus. In umlet hab ich auf jeden Fall nichts "fett" markiert...

Das scheint an umlet zu liegen. Wenn ich das Diagram mit umletino öffne, sieht das normal aus.

Reply:
Schräg. Naja, dann ist das halt so ;)

Reply:
Hab die beiden Diagramme nochmal mit dem vscode plugin re-exportiert, jetzt sehen die auch vernünftig aus.